game/layout/layoutloader.py

```diff
@@ -0,0 +1,203 @@
+            # the .load_file() method: 0:00:00.920409
+            temp_mis.load_file(miz)
+
+        for country in itertools.chain(
+            temp_mis.coalition["red"].countries.values(),
+            temp_mis.coalition["blue"].countries.values(),
+        ):
+            for dcs_group in itertools.chain(
+                temp_mis.country(country.name).vehicle_group,
+                temp_mis.country(country.name).ship_group,
+                temp_mis.country(country.name).static_group,
+            ):
+                try:
+                    mapping, group_id, group_mapping = self.mapping_for_group(
+                        mappings, dcs_group.name
+                    )
+                except KeyError:
+                    logging.warning(f"No mapping for dcs group {dcs_group.name}")
+                    continue
+
+                template = self._templates.get(mapping.name, None)
+                if template is None:
+                    # Create a new template
+                    template = TEMPLATE_TYPES[mapping.role](
+                        mapping.name, mapping.role, mapping.description
+                    )
+                    template.generic = mapping.generic
+                    template.tasks = mapping.tasks
+                    self._templates[template.name] = template
+
+                for i, unit in enumerate(dcs_group.units):
+                    group_template = None
+                    for group in template.groups:
+                        if group.name == group_mapping.name:
+                            # We already have a layoutgroup for this dcs_group
+                            group_template = group
+                    if not group_template:
+                        group_template = GroupLayout(
+                            group_mapping.name,
+                            [],
+                            group_mapping.group,
+                            group_mapping.unit_count,
+                            group_mapping.unit_types,
+                            group_mapping.unit_classes,
+                            group_mapping.alternative_classes,
+                        )
+                        group_template.optional = group_mapping.optional
+                        # Add the group at the correct position
+                        template.add_group(group_template, group_id)
+                    unit_template = LayoutUnit.from_unit(unit)
+                    if i == 0 and template.name not in template_position:
+                        template_position[template.name] = unit.position
+                    unit_template.position = (
+                        unit_template.position - template_position[template.name]
+                    )
+                    group_template.units.append(unit_template)
+
+    def by_name(self, template_name: str) -> Iterator[TheaterLayout]:
+        for template in self.layouts:
+            if template.name == template_name:
+                yield template
```

Comment:
This was already a dict, you don't need the linear search :) This also shouldn't return an `Iterator`, because there should only be one layout per name. Return `TheaterLayout` and `raise KeyError` if needed.